src/Components/RailSlider/Arrow.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { SFC } from "react"
+import styled from "styled-components"
+import Icon from "../Icon"
+
+interface Props {
+  direction: "left" | "right"
+  onClick?: () => void
+}
+
+export const Arrow: SFC<Props> = ({ direction, onClick }) => {
+  return (
+    <SliderArrow direction={direction} onClick={onClick}>
+      <Icon name={`chevron-${direction}` as any} color="black" fontSize="24px" />
+    </SliderArrow>
+  )
+}
+
+const SliderArrow = styled.div`
+  display: flex;
+  align-items: center;
+  position: absolute;
+  height: 100vh;
+  top: 0;
+  box-sizing: border-box;
+  ${(props: Props) => (props.direction === "left" ? "left: 0px;" : "")}
+  ${(props: Props) => (props.direction === "right" ? "right: 0px;" : "")}
```

Comment:
If you search Reaction for StyledFunction, you’ll find a way to write these components without needing to type props for each function.

Reply:
I was wondering about this, and the approach above is deliberate. Is there any benefit in going the other way? IMO `props: Props` is much less noisy and more succinct then using `StyledFunction` as it takes away two lines of code and the need for an intermediate variable.

Reply:
The benefit is that the call sites where this component is used gets code completion and is checked.

Some people have made more succinct suggestions here since mine, maybe one of these can work for us? https://github.com/styled-components/styled-components/issues/630

Reply:
Everything still seems to typecheck fine? 

<img width="461" alt="screen shot 2018-01-10 at 11 27 32 pm" src="https://user-images.githubusercontent.com/236943/34813218-efeb8084-f65d-11e7-905a-48a91c9930e7.png">

Reply:
But yeah, looking through the responses yours still seems to be the clearest out of the bunch.

Reply:
Oh that’s interesting! So either styled-components or TypeScript have made improvements.

We should revisit it then. Maybe something like this works as well nowadays which means you only need to type once?

```ts
const SliderArrow = styled.div<Props>`
  ...
`
```

Reply:
@alloy did you figure SC with TS out?

Reply:
> Maybe something like this works as well nowadays which means you only need to type once?

Unfortunately not, but maybe this makes sense to you? 

<img width="642" alt="screen shot 2018-01-12 at 11 30 03 am" src="https://user-images.githubusercontent.com/236943/34891867-025a653a-f78c-11e7-8aeb-cc83b007743c.png">